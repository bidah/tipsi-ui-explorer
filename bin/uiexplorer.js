#!/usr/bin/env node

/* eslint no-console: 0, no-var: 0, vars-on-top: 0, prefer-template: 0 */
var pkg = require('../package.json')
var path = require('path')
var shell = require('shelljs')
var cfonts = require('cfonts')
var program = require('commander')
var init = require('../cli/init')

program
  .version(pkg.version)
  .option('-h, --host <host>', 'host to listen on')
  .option('-p, --port <port>', 'port to listen on')
  .option('-c, --config-dir [dir-name]', 'uiexplorer config directory')
  .option('-i, --init', 'init uiexplorer in project')
  .option('-f, --force', 'force init')
  .parse(process.argv)


cfonts.say('UI Explorer', {
  font: 'simple3d',
  colors: ['candy'],
})

var projectDir = path.resolve()
var configDir = path.resolve(program.configDir || './uiexplorer')

if (program.init) {
  return init({ force: program.force })
}

shell.exec([
  'node node_modules/react-native/local-cli/cli.js start',
  '--projectRoots ' + configDir,
  '--root ' + projectDir,
  program.host ? '--host ' + program.host : '',
  program.port ? '--port ' + program.port : '',
].join(' '), { async: true })
