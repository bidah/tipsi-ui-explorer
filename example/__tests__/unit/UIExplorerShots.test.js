import { createShots } from 'tipsi-ui-explorer'
import '../../uiexplorer/examples'

createShots()
