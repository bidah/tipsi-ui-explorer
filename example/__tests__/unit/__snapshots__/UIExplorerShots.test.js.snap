exports[`UIExplorer api Alert Custom Buttons - Prop: onValueChange (Function) 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View
    style={
      Array [
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0C42FD",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }>
      Show alert with buttons
    </Text>
  </View>
</View>
`;

exports[`UIExplorer api Alert Simple Alert - Method: alert(title, message?, buttons?, options?, type?) 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View
    style={
      Array [
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0C42FD",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }>
      Show alert
    </Text>
  </View>
</View>
`;

exports[`UIExplorer components <Button /> Handle press - Prop: onPress (Function) 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View
    style={
      Array [
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0C42FD",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }>
      Press me!
    </Text>
  </View>
</View>
`;

exports[`UIExplorer components <Button /> Title - Prop: title (String) 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View
    style={
      Array [
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0C42FD",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }>
      Example
    </Text>
  </View>
</View>
`;

exports[`UIExplorer components <Switch /> Handle switch - Prop: onValueChange (Function) 1`] = `
<RCTSwitch
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      undefined,
    ]
  }
  value={false} />
`;

exports[`UIExplorer components <Switch /> Value - Prop: value (Boolean), set to true 1`] = `
<RCTSwitch
  disabled={false}
  onChange={[Function]}
  onPress={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      undefined,
    ]
  }
  value={true} />
`;
